subsys/usb/usb_device.c

```diff
@@ -233,12 +233,22 @@ static bool usb_handle_request(struct usb_setup_packet *setup,
  *
  * @return N/A
  */
-static void usb_data_to_host(void)
+static void usb_data_to_host(u16_t len)
 {
 	u32_t chunk = usb_dev.data_buf_residue;
 
+#if CONFIG_USB_NRF52840
+	if ((chunk == USB_MAX_CTRL_MPS) && (len > USB_MAX_CTRL_MPS)) {
```

Comment:
If the chunk=64 but there are more data to be send it would send ZLP after first packet, should we check `usb_dev.data_buf_residue`